packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -316,6 +320,7 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
   ///
   /// If drag is only allowed along a defined axis, this value may be negative to
   /// differentiate the direction of the drag.
+  double get globalDistanceMoved => _globalDistanceMoved;
   late double _globalDistanceMoved;
```

Comment:
I feel about 80% confident that this is a fantastic improvement to the Flutter framework, but a small concern is that once something becomes public, the potential amount of future breaking changes increases.

What is the advantage of subclassing `DragGestureRecognizer` rather than extending `OneSequenceGestureRecognizer` directly?

Reply:
It's true that `OneSequenceGestureRecognizer` can be extended directly. However, for small changes, such as the one I described in the PR description, where we just want to customize the `PanGestureRecognizer` a little bit. 

It looks overwhelming to me that we need to copy the whole `DragGestureRecognizer` just to tweak `PanGestureRecognizer` a little bit.

What do you think?

Reply:
Yep, that sounds fair. Thanks!